apps/src/gamelab/blocks.js

```diff
@@ -282,6 +285,65 @@ export default {
       blockText: 'hide title screen',
       args: [],
     });
+
+    // Legacy style block definitions :(
+    const generator = blockly.Generator.get('JavaScript');
+
+    Blockly.Blocks.sprite_variables_get = {
+      // Variable getter.
+      init: function () {
+        var fieldLabel = new Blockly.FieldLabel(Blockly.Msg.VARIABLES_GET_ITEM);
+        // Must be marked EDITABLE so that cloned blocks share the same var name
+        fieldLabel.EDITABLE = true;
+        this.setHelpUrl(Blockly.Msg.VARIABLES_GET_HELPURL);
+        this.setHSV(131, 0.64, 0.62);
+        this.appendDummyInput()
+            .appendTitle(Blockly.Msg.VARIABLES_GET_TITLE)
+            .appendTitle(Blockly.disableVariableEditing ? fieldLabel
+                : new Blockly.FieldVariable(Blockly.Msg.VARIABLES_GET_ITEM), 'VAR')
+            .appendTitle(Blockly.Msg.VARIABLES_GET_TAIL);
+        this.setStrictOutput(true, Blockly.BlockValueType.SPRITE);
+        this.setTooltip(Blockly.Msg.VARIABLES_GET_TOOLTIP);
+      },
+      getVars: function () {
+        return [this.getTitleValue('VAR')];
+      },
+      renameVar: function (oldName, newName) {
+        if (Blockly.Names.equals(oldName, this.getTitleValue('VAR'))) {
+          this.setTitleValue(newName, 'VAR');
+        }
+      },
+    };
+    generator.sprite_variables_get = generator.variables_get;
+
+    Blockly.Blocks.sprite_variables_set = {
+      // Variable setter.
+      init: function () {
+        var fieldLabel = new Blockly.FieldLabel(Blockly.Msg.VARIABLES_SET_ITEM);
+        // Must be marked EDITABLE so that cloned blocks share the same var name
+        fieldLabel.EDITABLE = true;
+        this.setHelpUrl(Blockly.Msg.VARIABLES_SET_HELPURL);
+        this.setHSV(131, 0.64, 0.62);
+        this.appendValueInput('VALUE')
+            .setStrictCheck(Blockly.BlockValueType.SPRITE)
+            .appendTitle(Blockly.Msg.VARIABLES_SET_TITLE)
+            .appendTitle(Blockly.disableVariableEditing ? fieldLabel
+              : new Blockly.FieldVariable(Blockly.Msg.VARIABLES_SET_ITEM), 'VAR')
+            .appendTitle(Blockly.Msg.VARIABLES_SET_TAIL);
+        this.setPreviousStatement(true);
+        this.setNextStatement(true);
+        this.setTooltip(Blockly.Msg.VARIABLES_SET_TOOLTIP);
+      },
+      getVars: function () {
+        return [this.getTitleValue('VAR')];
+      },
+      renameVar: function (oldName, newName) {
+        if (Blockly.Names.equals(oldName, this.getTitleValue('VAR'))) {
+          this.setTitleValue(newName, 'VAR');
+        }
+      },
+    };
+    generator.sprite_variables_set = generator.variables_set;
```

Comment:
Would be good to de-dupe this against the `variables_get` and `variables_set` block definitions.

Reply:
Hmm, I can probably add a variable block creator in blockly that would also handle `parameters_get`. I've got one more change I need to make to sprite blocks to get separate lists of variables, but after that I'll look into deduping.